src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -170,6 +174,18 @@ double CheckGradient(FunctionType& function, const double eps = 1e-7)
 
     // Compute numerical gradients using the costs calculated above.
     estGradient(i) = (costPlus - costMinus) / (2 * eps);
+    if (k){
+      std::cout << i + 1 << "th Iteration:" << std::endl;
+      std::cout << "Weight: " << tmp << std::endl;
+      std::cout << "CostPlus: " << costPlus << std::endl;
+      std::cout << "CostMinus: " << costMinus << std::endl;
+      std::cout << "Established Gradient: " << estGradient(i) << std::endl << std::endl;
```

Comment:
@zoq I ran these changes over the iterations for the `BilinearInterpolation` Layer. It turns out that `costPlus` and `costMinus` values return the same `double` value, and  hence the `estGradient` becomes `0` for each output point.

Reply:
I see, since the gradient is always one , I think there is no need to add the test here.